### DISCUSSION TITLE
How can I make my own structure?

### DISCUSSION THREAD

**Discussion by hurmantrout** [preactjs/preact-cli#1601] [Q&A]
Hey guys,
for some reasons I don't wanna use "preact-cli" in my project.
so, How can I make my own structure with support of (Webpack, Hot-loading, BabelJS, CSS imports, Typescript, Preact, Preact Router, ...).
I'd be grateful if you could help :)

**Comment by hurmantrout** [comment]
+ Production Build & Development Mode

**Comment by rschristian** [comment]
Is the reason that you don't want to use Preact-CLI related to a bug or missing feature?

You can certainly look at our config but it's a fair bit if you've never dealt with Webpack before. I'm sure tutorials exist out there for this sort of thing.

**Reply by hurmantrout** [reply]
thanks for recommending webpack, i'll surely try it.
and no, preact-cli is **great**.
i just wanna make my project lighter and lighter!

thanks for your help :)

**Reply by rschristian** [reply]
There is [WMR](https://github.com/preactjs/wmr) as well, if you weren't aware.
